lib/rest.js

```diff
@@ -70,8 +70,10 @@ module.exports = class Rest {
             }
             if (
                 Array.isArray(collector[iteratorField]) &&
-                collector[iteratorField].length >= temp.count
+                collector[iteratorField].length >= temp.count &&
+                temp.count != temp.pageSize
```

Comment:
if pagesize is 25 and there is actually 25 results - not great outcomes

Reply:
Well, then we have 1 call with 0 results. Give me another option.

It's a workaround, not a fix. Only stmc product can fix it

Reply:
I would rather hard code a check which only runs for transactionalmessage then - also put a support ticket id in :) then we can only apply in this one case, even if it is a hard coded exception

Reply:
updated according to your idea. Pls review again